node/src/actors/session/handlers.rs

```diff
@@ -116,14 +116,36 @@ impl StreamHandler<BytesMut, Error> for Session {
                     (
                         _,
                         SessionStatus::Unconsolidated,
-                        Command::Version(Version { sender_address, .. }),
+                        Command::Version(Version {
+                            sender_address,
+                            timestamp,
+                            ..
+                        }),
                     ) => {
-                        let msgs = handshake_version(self, &sender_address);
-                        for msg in msgs {
-                            self.send_message(msg);
+                        let current_ts = get_timestamp();
+                        let received_ts = timestamp;
+
+                        if self.handshake_max_ts_diff != 0
+                            && (current_ts - received_ts).abs() > self.handshake_max_ts_diff
+                        {
+                            log::warn!(
+                                "Dropping peer because their timestamp is too big (±{} seconds).\n\
+                                 Current timestamp: {}\nReceived timestamp: {}",
+                                self.handshake_max_ts_diff,
```

Comment:
I meant to use actual difference here:

```suggestion
                                (current-ts - received_ts),
```

Reply:
sure, I am sleepy today >_<